### PR TITLE
allow control over service type

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

File: pkg/resources/services/service.go [hunk 1]
```diff
@@ -72,14 +77,18 @@ func NewServiceForCR(m *v1alpha1.Interconnect, requestCert bool) *corev1.Service
...
 		Spec: corev1.ServiceSpec{
-			Type:     "LoadBalancer",
 			Selector: labels,
 			Ports:    servicePortsForRouter(m),
 		},
 	}
 	if requestCert {
 		service.Annotations = map[string]string{constants.CertRequestAnnotation: m.Name + "-cert"}
 	}
+	if m.Spec.DeploymentPlan.ServiceType == "ClusterIP" || m.Spec.DeploymentPlan.ServiceType == "NodePort" || m.Spec.DeploymentPlan.ServiceType == "LoadBalancer"  {
```
Comment (ajssmith) [inline pkg/resources/services/service.go]: lgtm, two questions:

1. When ServiceType is not provided, what is the default?
2. When ServiceType == "ClusterIP" should Spec.ClusterIP be set to "None"?

Comment (ajssmith) [inline pkg/resources/services/service.go]: Needs a go fmt update

File: pkg/apis/interconnectedcloud/v1alpha1/interconnect_types.go [hunk 2]
```diff
@@ -105,6 +105,7 @@ type DeploymentPlanType struct {
 	Resources    corev1.ResourceRequirements `json:"resources,omitempty"`
 	Issuer       string                      `json:"issuer,omitempty"`
 	LivenessPort int32                       `json:"livenessPort,omitempty"`
+	ServiceType  string                      `json:"serviceType,omitempty"`
```
Comment (ajssmith) [inline pkg/apis/interconnectedcloud/v1alpha1/interconnect_types.go]: Change needs update to:
deploy/olm-catalog/qdr-operator/0.1.0/catalog-source.yaml
pkg/apis/interconnectedcloud/v1alpha1/zz_generated.deepcopy.go

File: pkg/resources/services/service.go (hunk 1 above)
Comment (grs) [reply pkg/resources/services/service.go]: 1. ClusterIP is the default.
2. Spec.ClusterIP isn't set at all; when type is ClusterIP the controller will pick an IP and set that field. (Not sure if I am answering the right question?)

Comment (grs) [reply pkg/resources/services/service.go]: done

File: pkg/apis/interconnectedcloud/v1alpha1/interconnect_types.go (hunk 2 above)
Comment (grs) [reply pkg/apis/interconnectedcloud/v1alpha1/interconnect_types.go]: catalog source has been regenerated.

What should trigger the deepcopy update? A full make seems not to make any change.

Comment (grs) [reply pkg/apis/interconnectedcloud/v1alpha1/interconnect_types.go]: Looks like the simple typed fields aren't explicitly copied, but rely on the struct level assignment, so this change I think is not expected to regenerate the method.

**Pull request merged**